### PR TITLE
update recaptcha - ignore home page

### PR DESCRIPTION
**Pull request by Leowisd**
This implements https://github.com/oncokb/oncokb/issues/2361

 Disable recaptcha on home page can reduce the possibility of showing challenge in incognito model.

### CODE REVIEW

File: src/main/webapp/app/shared/auth/RecaptchaBoundaryRoute.tsx [hunk 1]
```diff
@@ -1,31 +1,51 @@
...
-          .
-        </div>
-      </>
-    );
+@observer
+export class RecaptchaBoundaryRoute extends React.Component<
+  IRecaptchaBoundaryRoute
+> {
+  @computed
+  get recaptchaVerifyed() {
```
Comment (zhx828) [inline src/main/webapp/app/shared/auth/RecaptchaBoundaryRoute.tsx:19]: Why are we computing on a variable?

File: src/main/webapp/app/store/WindowStore.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ export interface IWindowSize {
 class WindowStore {
   @observable size: IWindowSize;
   public recaptchaRef: any;
+  @observable recaptchaVerified: boolean;
```
Comment (zhx828) [inline src/main/webapp/app/store/WindowStore.ts:14]: The goal is to verify one time and works for the rest of the session?

**Review by zhx828 — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/shared/auth/RecaptchaBoundaryRoute.tsx (hunk 1 above)
Comment (Leowisd) [reply src/main/webapp/app/shared/auth/RecaptchaBoundaryRoute.tsx:19]: This route component should be re-render to show real page component when the first time recaptcha got verified, or the page will only show warning message.

File: src/main/webapp/app/store/WindowStore.ts (hunk 2 above)
Comment (Leowisd) [reply src/main/webapp/app/store/WindowStore.ts:14]: Yes.

**Review by zhx828 — APPROVED** [review]

**Pull request merged**